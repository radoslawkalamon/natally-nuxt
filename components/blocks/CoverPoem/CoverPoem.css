.cover-poem {
  background-repeat: repeat;
  border-radius: var(--border-radius);
  height: 72px;
  max-width: var(--max-width-body);
  width: 100%;
}
