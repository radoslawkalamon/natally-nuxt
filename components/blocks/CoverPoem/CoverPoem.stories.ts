import type { Meta, Story } from '@storybook/vue'
import BlocksCoverPoem from '@/components/blocks/CoverPoem/CoverPoem.vue'

const meta: Meta = {
  title: 'Blocks / Cover Poem',
  component: BlocksCoverPoem
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: {
    BlocksCoverPoem
  },
  props: Object.keys(argTypes),
  template: '<BlocksCoverPoem v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  images: ['https://www.fillmurray.com/100/150']
}
