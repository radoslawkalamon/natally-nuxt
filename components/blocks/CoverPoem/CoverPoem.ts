import Vue from 'vue'

export default Vue.extend({
  name: 'BlocksCoverPoem',
  components: {
    ComponentsCover: () => import(/* webpackChunkName: "components" */'@/components/components/Cover/Cover.vue')
  },
  props: {
    images: {
      type: Array as Vue.PropType<String[]>,
      required: true
    }
  }
})
