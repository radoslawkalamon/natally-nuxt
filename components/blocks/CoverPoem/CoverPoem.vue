<template>
  <ComponentsCover
    class="cover-poem"
    :images="images"
  />
</template>

<script src="./CoverPoem.ts" lang="ts"></script>
<style src="./CoverPoem.css" scoped></style>
