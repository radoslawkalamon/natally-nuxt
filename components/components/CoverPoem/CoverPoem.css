.cover-poem {
  background-color: var(--color-betwixt);
  background-position: center;
  background-repeat: repeat;
  border-radius: var(--border-radius);
  height: 72px;
  max-width: var(--max-width-body);
  width: 100%;
}
