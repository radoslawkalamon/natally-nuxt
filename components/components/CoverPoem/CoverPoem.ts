import Vue from 'vue'

export default Vue.extend({
  name: 'ComponentsCoverPoem',
  props: {
    image: {
      type: String,
      required: true
    }
  },
  computed: {
    style () {
      return {
        backgroundImage: `url('${this.image}')`
      }
    }
  }
})
