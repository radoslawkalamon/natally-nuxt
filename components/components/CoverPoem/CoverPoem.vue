<template v-once>
  <div
    aria-hidden="true"
    class="cover-poem"
    data-test="components-cover-poem"
    :style="style"
  />
</template>

<script src="./CoverPoem.ts" lang="ts"></script>
<style src="./CoverPoem.css" scoped></style>
