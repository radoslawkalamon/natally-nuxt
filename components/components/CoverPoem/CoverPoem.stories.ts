import { Meta, Story } from '@storybook/vue'
import CoverPoem from './CoverPoem.vue'

const meta: Meta = {
  title: 'Components / Cover Poem',
  component: CoverPoem
}
export default meta

const Template: Story = (_args, { argTypes }) => ({
  components: { CoverPoem },
  props: Object.keys(argTypes),
  template: '<CoverPoem v-bind="$props" />'
})

export const Default: Story = Template.bind({})
Default.args = {
  image: 'https://www.fillmurray.com/100/150'
}
