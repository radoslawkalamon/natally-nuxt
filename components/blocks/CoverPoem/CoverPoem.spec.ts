import CoverPoem from '@/components/blocks/CoverPoem/CoverPoem.vue'
import { shallPassIntegrationSanityTest, shallPassUnitSanityTest } from '@/devtools/jest.common.spec'

describe('Blocks / Cover Poem', () => {
  describe('Unit', () => {
    shallPassUnitSanityTest({
      component: CoverPoem,
      description: 'no images',
      options: {
        propsData: {
          images: []
        },
        stubs: ['ComponentsCover']
      }
    })

    shallPassUnitSanityTest({
      component: CoverPoem,
      description: 'with images',
      options: {
        propsData: {
          images: ['/images/post.test-post.1x.jpg']
        },
        stubs: ['ComponentsCover']
      }
    })
  })

  describe('Integration', () => {
    shallPassIntegrationSanityTest({
      component: CoverPoem,
      description: 'no images',
      options: {
        propsData: {
          images: []
        }
      }
    })

    shallPassIntegrationSanityTest({
      component: CoverPoem,
      description: 'with images',
      options: {
        propsData: {
          images: ['/images/post.test-post.1x.jpg']
        }
      }
    })
  })
})
