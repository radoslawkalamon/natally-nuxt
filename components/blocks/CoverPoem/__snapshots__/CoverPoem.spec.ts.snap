// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks / Cover Poem Integration shall pass integration sanity test | no images 1`] = `"<div aria-hidden=\\"true\\" class=\\"cover cover-poem\\"></div>"`;

exports[`Blocks / Cover Poem Integration shall pass integration sanity test | with images 1`] = `"<div aria-hidden=\\"true\\" class=\\"cover cover-poem\\"></div>"`;

exports[`Blocks / Cover Poem Unit shall pass unit sanity test | no images 1`] = `"<componentscover-stub images=\\"\\" class=\\"cover-poem\\"></componentscover-stub>"`;

exports[`Blocks / Cover Poem Unit shall pass unit sanity test | with images 1`] = `"<componentscover-stub images=\\"/images/post.test-post.1x.jpg\\" class=\\"cover-poem\\"></componentscover-stub>"`;
